docs/develop/dotnet/set-up.mdx

```diff
@@ -0,0 +1,550 @@
+}
+}`}
+
+</CodeSnippet>
+</>
+}>
+
+### Create the Workflow
+
+Workflows orchestrate Activities and contain the application logic. Temporal Workflows are resilient. They can run—and keeping running—for years, even if the underlying infrastructure fails. If the application itself crashes, Temporal will automatically recreate its pre-failure state so it can continue right where it left off.
+
+Create a file called `SayHelloWorkflow.cs`:
+
+</SetupStep>
+
+<SetupStep code={
+<>
+
+<p><strong>Create a Worker file (Program.cs):</strong></p>
+<CodeSnippet language="csharp">
+{`using MyNamespace;
+using Temporalio.Client;
+using Temporalio.Worker;
+
+// Create a client to localhost on "default" namespace
+var client = await TemporalClient.ConnectAsync(new("localhost:7233"));
+
+// Cancellation token to shutdown worker on ctrl+c
+using var tokenSource = new CancellationTokenSource();
+Console.CancelKeyPress += (\_, eventArgs) =>
+{
+tokenSource.Cancel();
+eventArgs.Cancel = true;
+};
+
+// Create an activity instance since we have instance activities. If we had
+// all static activities, we could just reference those directly.
+var activities = new MyActivities();
+
+// Create worker with the activity and workflow registered
+using var worker = new TemporalWorker(
+client,
+new TemporalWorkerOptions("my-task-queue").
+AddActivity(activities.SayHello).
+AddWorkflow<SayHelloWorkflow>());
+
+// Run worker until cancelled
+Console.WriteLine("Running worker");
+try
+{
+await worker.ExecuteAsync(tokenSource.Token);
+}
+catch (OperationCanceledException)
+{
+Console.WriteLine("Worker cancelled");
+}`}
+
+</CodeSnippet>
+
+    <p><strong>Run the Worker:</strong></p>
+    <CodeSnippet language="bash">
+    dotnet run
+    </CodeSnippet>
+
+</>
+}>
+
+### Create and Run the Worker
+
+With your Activity and Workflow defined, you need a Worker to execute them. A Worker polls a Task Queue, that you configure it to poll, looking for work to do. Once the Worker dequeues the a Workflow or Activity task from the Task Queue, it then executes that task.
+
+Workers are a crucial part of your Temporal application as they're what actually execute the tasks defined in your Workflows and Activities.
+For more information on Workers, see [Understanding Temporal](/evaluate/understanding-temporal#workers) and a [deep dive into Workers](/workers).
+
+Create a file called `Program.cs`:
+
+Keep this terminal running - you should see "Running worker" displayed.
+
+</SetupStep>
+
+<SetupStep code={
+<>
+
+<div className="execution-options">
+  <h4>Choose Your Execution Method</h4>
+  <p>With the Worker running, you can start a Workflow in two ways:</p>
+</div>
+
+    <div className="option-section">
+      <h5>Option 1: Using Temporal CLI (Recommended for testing)</h5>
+      <p>Open a new terminal and run:</p>
+      <CodeSnippet language="bash">
+      temporal workflow start --type SayHelloWorkflow --task-queue my-task-queue --input '"Temporal"'
```

Comment:
Getting this error when I run this command: `time=2025-07-17T11:17:50.801 level=ERROR msg="failed reaching server: context deadline exceeded"`

Reply:
This fixed the error: `temporal env set default.address 127.0.0.1:7233`